### PR TITLE
Login Reminder: log notification dismiss event

### PR DESCRIPTION
**Pull request by jaclync**
Part of #7318 

### Description

When I first implemented local notifications, I wasn't aware that we can also handle the "dismiss" action until I stumbled upon `UNNotificationDismissActionIdentifier` and `UNNotificationDefaultActionIdentifier` when creating mocks for `UNNotificationResponse` in unit tests.

This PR updated the local notification handling logic to use `UNNotificationDismissActionIdentifier` and `UNNotificationDefaultActionIdentifier` to handle "dismiss" and "default" actions. For "dismiss" action, `_login_local_notification_dismissed` event is tracked.

#### Dismiss action

Please update the 24 hours schedule to a few seconds or a minute in `AuthenticationManager` (search for "86400") to test this PR.

- Reinstall the app to clear any previous notifications permission state
- Skip the onboarding if needed
- Tap "Enter Your Store Address"
- Enter an invalid store address or a store address that isn't a WP site --> an error screen should be shown
- Quickly put the app to the background since we don't support local notifications in the foreground for this iteration --> a local notification should be shown after the duration you set in `AuthenticationManager`
- Swipe the notification and tap on "Clear" --> the notification should disappear, and an event should be logged `🔵 Tracked login_local_notification_dismissed, properties: [AnyHashable("type"): "site_address_error"]`

#### Default action

- Continue from the previous test case
- Tap "Enter Your Store Address"
- Enter an invalid store address or a store address that isn't a WP site --> an error screen should be shown
- Quickly put the app to the background since we don't support local notifications in the foreground for this iteration --> a local notification should be shown after the duration you set in `AuthenticationManager`
- Tap on the notification itself --> it should open the app, and an event should be logged `🔵 Tracked login_local_notification_tapped, properties: [AnyHashable("action"): "default", AnyHashable("type"): "site_address_error"]`

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerceTests/AppCoordinatorTests.swift [hunk 1]
```diff
@@ -203,6 +203,78 @@ final class AppCoordinatorTests: XCTestCase {
         // Then
         XCTAssertEqual(analytics.receivedEvents, [WooAnalyticsStat.loginOnboardingShown.rawValue])
     }
+
+    // MARK: - Login reminder analytics
```
Comment (selanthiraiyan) [inline WooCommerce/WooCommerceTests/AppCoordinatorTests.swift:207]: Nice tests. ❤️

File: WooCommerce/Classes/Notifications/PushNotificationsManager.swift [hunk 2]
```diff
@@ -322,7 +322,10 @@ extension PushNotificationsManager {
                                                       intentIdentifiers: [],
                                                       hiddenPreviewsBodyPlaceholder: nil,
                                                       categorySummaryFormat: nil,
-                                                      options: .allowAnnouncement)
+                                                      // `customDismissAction` option is required for the dismiss action callback in
+                                                      // `UNUserNotificationCenterDelegate.userNotificationCenter(_:didReceive:)`
+                                                      // with action identifier `UNNotificationDismissActionIdentifier`.
+                                                      options: .customDismissAction)
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Notifications/PushNotificationsManager.swift:328]: Thank you for adding comments! 💯

File: WooCommerce/Classes/ViewRelated/AppCoordinator.swift [hunk 3]
```diff
@@ -250,7 +253,7 @@ private extension AppCoordinator {
                 "action": "contact_support",
                 "type": response.notification.request.identifier
             ])
-        default:
+        case UNNotificationDefaultActionIdentifier:
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/AppCoordinator.swift:256]: It feels like this `func handleLocalNotificationResponse` method is growing large to handle many types of local notifications. If you feel the same way, perhaps we should try to think of a way to break this down into multiple private methods. (In a future PR 😅 )

**Review by selanthiraiyan — APPROVED** [review]
It works as expected. 🎉 

:shipit:

File: WooCommerce/Classes/ViewRelated/AppCoordinator.swift (hunk 3 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/AppCoordinator.swift:256]: Thanks for the suggestion, I'll keep this in mind the next time touching this switch statement 🙂

**Pull request merged**